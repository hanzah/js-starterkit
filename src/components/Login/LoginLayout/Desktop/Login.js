import React, { Component, PropTypes } from 'react';

import styles from './Login.scss';
import inlineStyle from './LoginStyle';

const {object} = PropTypes;

class Login extends Component {
  render() {
    return (
      <div className={styles.login_background}>
        <div className={styles.bckg_image} style={inlineStyle.bckg}>
          <div className={styles.login_container}>
            <div className={styles.login_logo} data-spec='login-logo'>
              {this.props.logo}
            </div>
            <div className={styles.login_form} data-spec='login-form'>
              {this.props.form}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  logo: object,
  form: object
}

export default Login;