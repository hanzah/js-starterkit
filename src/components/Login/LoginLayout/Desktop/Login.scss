.login_background {
  background-image: -moz-linear-gradient( 45deg, rgb(5, 104, 174) 0%, rgb(0, 159, 219) 100%);
  background-image: -webkit-linear-gradient( 45deg, rgb(5, 104, 174) 0%, rgb(0, 159, 219) 100%);
  background-image: -ms-linear-gradient( 45deg, rgb(5, 104, 174) 0%, rgb(0, 159, 219) 100%);
  height: 100vh;
  width: 100%;
  .bckg_image {
    height: 100vh;
    width: 100%;
  }
}

.login_container {
  display: flex;
  height: 100%;
  width: 100%;
  align-items: center;
  justify-content: center;
}

.login_logo {
  display: flex;
  width: 350px;
  height: 450px;
  background-color: rgba(255, 255, 255, 0.2);
  border-radius: 5px 0px 0px 5px;
  position: relative
}

.login_form {
  display: flex;
  width: 350px;
  height: 450px;
  background-color: rgb(255, 255, 255);
  border-radius: 0px 5px 5px 0px;
  position: relative;
  >div {
    width: 100%;
  }
}