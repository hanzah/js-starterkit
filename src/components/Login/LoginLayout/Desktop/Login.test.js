import React from 'react';
import { mount, shallow } from 'enzyme';

import Login from './Login';

describe('<Login/>', () => {
  it('should render a component', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });
});