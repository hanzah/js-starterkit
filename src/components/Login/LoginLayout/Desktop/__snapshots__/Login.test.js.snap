exports[`<Login/> should render a component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}>
    <div
        className={undefined}
        style={
            Object {
                "backgroundImage": "url(test-file-stub)",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "cover",
              }
        }>
        <div
            className={undefined}>
            <div
                className={undefined}
                data-spec="login-logo" />
            <div
                className={undefined}
                data-spec="login-form" />
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className={undefined}>
      <div
            className={undefined}
            style={
                  Object {
                        "backgroundImage": "url(test-file-stub)",
                        "backgroundRepeat": "no-repeat",
                        "backgroundSize": "cover",
                      }
            }>
            <div
                  className={undefined}>
                  <div
                        className={undefined}
                        data-spec="login-logo" />
                  <div
                        className={undefined}
                        data-spec="login-form" />
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Login />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Login {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}>
          <div
                    className={undefined}
                    style={
                              Object {
                                        "backgroundImage": "url(test-file-stub)",
                                        "backgroundRepeat": "no-repeat",
                                        "backgroundSize": "cover",
                                      }
                    }>
                    <div
                              className={undefined}>
                              <div
                                        className={undefined}
                                        data-spec="login-logo" />
                              <div
                                        className={undefined}
                                        data-spec="login-form" />
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}>
          <div
                    className={undefined}
                    style={
                              Object {
                                        "backgroundImage": "url(test-file-stub)",
                                        "backgroundRepeat": "no-repeat",
                                        "backgroundSize": "cover",
                                      }
                    }>
                    <div
                              className={undefined}>
                              <div
                                        className={undefined}
                                        data-spec="login-logo" />
                              <div
                                        className={undefined}
                                        data-spec="login-form" />
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Login />,
}
`;
